src/test/scala/query/AssignmentSpec.scala

```diff
@@ -0,0 +1,80 @@
+package com.originate.scalypher.test.query
+
+import com.originate.scalypher.types.ReferenceableMap
+import com.originate.scalypher.Property
+import com.originate.scalypher.Label
+import com.originate.scalypher.path.AnyNode
+import com.originate.scalypher.path.AnyRelationship
+import com.originate.scalypher.RemovePropertyAssignment
+import com.originate.scalypher.MergePropertiesAssignment
+import com.originate.scalypher.OverwriteAssignment
+import com.originate.scalypher.PropertyAssignment
+
+import org.scalatest._
+
+class AssignmentSpec extends WordSpec with Matchers {
+
+  val startNode = AnyNode()
+  val endNode = AnyNode()
+  val relationship = AnyRelationship()
+
+  val label1 = Label("label1")
+  val label2 = Label("label2")
+  val properties = Seq(
+    Property("name", "matt"),
+    Property("age", 12)
+  )
+
+  val map: ReferenceableMap = Map(
+    startNode -> "a1",
+    endNode -> "a2",
+    relationship -> "a3"
+  )
+
+  "removing properties" must {
+
+    "set elements to null" in {
+      val assignment = startNode.property("name").assignNull
+      assignment.toQuery(map) shouldBe "a1.name = NULL"
+    }
+
+  }
+
+  "assigning properties" must {
+
+    "set elements to strings" in {
+      val assignment = startNode.property("name") assign "asdf"
```

Comment:
call me a pascal programmer but I'd prefer to read this as
```
startNode.property("name") := "asdf"
```

i'd also settle for `is`: `startNode.property("name") is "asdf"`. `assign` is a little awkward, but the end choice ties pretty closely to what a neo4j expression means. declarative of the world (like my suggestions imply) or declarative an operation (which `assign` implies). if it's the latter, i still think there's a better name, but it escapes me